src/napari_spatialdata/_interactive.py

```diff
@@ -163,41 +201,44 @@ def _add_label(self, key: str) -> None:
                     self._sdata.table.obs[self._sdata.table.uns["spatialdata_attrs"]["region_key"]] == key
                 ],
                 "labels_key": self._sdata.table.uns["spatialdata_attrs"]["instance_key"],
+                "active_in_cs": {selected_cs},
```

Comment:
Do I understand this correctly that we are assuming that whenever "_add_label()" is called, the new labels layer is active? Are there cases where this might not be true? I think we should probably make a note of this in the docstring. This could be a uprising side effect down the road.

(this would hold for the other `_add_*` methods)

Reply:
At the moment we only directly add layers by using the functions in the sdata widget. These ultimately give a call to viewer.add_<layer_type> for which no parameter `visible` is used now. 

I think what we need and which will be part of a follow up PR (my next one or second next) for also saving shapes / points is a spatialdata viewer which holds a reference to the napari viewer or inherits from it so we can have some additional callbacks. Here we could add an option instead of `add_image` and so on to `add_spatialelement_layer` which is ultimately `add_layer` with some extra bells and whistles.

Reply:
With the follow up PR for saving points / shapes I add the current coordinate system metadata to indicate in which coordinate system the layer lives. When then adding a new layer by clicking on one of the napari buttons to add the layer, it checks the metadata of the layer which was active to inherit metadata regarding coordinate system and spatialdata object. 
While technically we could add these custom callbacks in the sdata widget, it feels convoluted to me so that is why I would go for the viewer solution I talked about in the comment above.